test/scripting/stable_save.lua

```diff
@@ -1,13 +1,30 @@
 include "scripting/coroutine.lua"
 
 -- Save the game so that reloading does not skip
-function stable_save(game, savename)
-   local old_speed = game.desired_speed
+function stable_save(game, savename, desired_speed)
+   local mapview = wl.ui.MapView()
+   local last_save_time = game.last_save_time
+   print("###### stable_save: last save time " .. last_save_time)
+
    game.desired_speed = 1000
    sleep(100)
    game:save(savename)
    game.desired_speed = 1000
-   sleep(2000)  -- Give the loaded game a chance to catch up
-   game.desired_speed = old_speed
-   sleep(1000)
+
+   -- Wait until save was finished
+   local counter = 0
+   repeat
+      sleep(200)
+   until game.last_save_time ~= last_save_time or counter > 20
+   print("###### stable_save: new save time " .. game.last_save_time)
+
+   -- Give the loaded game a chance to catch up
+   counter = 0
+   while mapview.average_fps < 20 and counter < 100 do
+      sleep(200)
+      counter = counter + 1
```

Comment:
This pauses execution of the script for up to 20 seconds, but the game continues running. Can we be sure that nothing testcase-critical will happen during this undefined waiting period?

Reply:
We already had a sleep here before, just for a fixed, completely arbitrary time. It should be OK as long as we don't launch this function in a coroutine.

I ran the test suite on the buggy setup for about 10 times without any failures, so it should be OK.